src/Tests/Microsoft.Diagnostics.Monitoring.Tool.FunctionalTests/StacksTests.cs

```diff
@@ -0,0 +1,198 @@
+                        }
+                        if ((line == expectedFrames.First()) || (actualFrames.Count > 0))
+                        {
+                            actualFrames.Add(line);
+                        }
+                    }
+
+                    Assert.Equal(expectedFrames, actualFrames);
+
+                    await runner.SendCommandAsync(TestAppScenarios.Stacks.Commands.Continue);
+                });
+        }
+
+        [Fact]
+        public Task TestJsonStacks()
+        {
+            return ScenarioRunner.SingleTarget(
+                _outputHelper,
+                _httpClientFactory,
+                WebApi.DiagnosticPortConnectionMode.Listen,
+                TestAppScenarios.Stacks.Name,
+                appValidate: async (runner, client) =>
+                {
+                    int processId = await runner.ProcessIdTask;
+
+                    using ResponseStreamHolder holder = await client.CaptureStacksAsync(processId, plainText: false);
+                    Assert.NotNull(holder);
+
+                    WebApi.Models.StackResult result = await JsonSerializer.DeserializeAsync<WebApi.Models.StackResult>(holder.Stream);
+
+                    var actualFrames = new List<WebApi.Models.StackFrame>();
+                    WebApi.Models.StackFrame[] expectedFrames = new WebApi.Models.StackFrame[]
+                    {
+                        new WebApi.Models.StackFrame
+                        {
+                            ModuleName = ExpectedModule,
+                            ClassName = ExpectedClass,
+                            MethodName = ExpectedCallbackFunction
+                        },
+                        new WebApi.Models.StackFrame
+                        {
+                            ModuleName = NativeFrame,
+                            ClassName = NativeFrame,
+                            MethodName = NativeFrame
+                        },
+                        new WebApi.Models.StackFrame
+                        {
+                            ModuleName = ExpectedModule,
+                            ClassName = ExpectedClass,
+                            MethodName = ExpectedFunction
+                        }
+                    };
+
+                    foreach (WebApi.Models.Stack stack in result.Stacks)
+                    {
+                        actualFrames.Clear();
+                        foreach(var frame in stack.Frames)
+                        {
+                            if (actualFrames.Count == 3)
+                            {
+                                break;
+                            }
+                            if (AreFramesEqual(expectedFrames.First(), frame) || actualFrames.Count > 0)
+                            {
+                                actualFrames.Add(frame);
+                            }
+                        }
+                    }
+
+                    Assert.Equal(expectedFrames.Length, actualFrames.Count);
```

Comment:
Since this is outside the `foreach` loop, aren't we only checking the `expectedFrames` against the last `stack` (given that `actualFrames` gets cleared for every iteration)?

Reply:
I'm only expecting the matching frame to be on one of the stacks. I basically iterate through all the stacks, finding the first matching frame. Then I take 2 more frames and then validate that all the frames are what I'm expecting them to be.